File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {

    int student[32];
    student[0] = 0;
    student[n+1] = 0;

    for(int i=1;i<=n;i++){student[i] = 1;}
    for(int i=0;i<lost.size();i++){student[lost[i]] = 0;}
    for(int i=0;i<reserve.size();i++){student[reserve[i]] += 1;}

    int answer = 0;
    for(int i=1;i<=n;i++){
        if(student[i]>0){answer++;}
        else if(student[i-1] == 2){answer++;}
        else if(student[i+1] == 2){answer++;student[i+1] =1;}
    }
        
    return answer;
}

File: 프로그래머스/1/42862.체육복-Python3/체육복.py
def solution(n, lost, reserve):
    answer = 0
    counts = [1]*(n+2)
    counts[0] = 0
    counts[n+1] = 0
    
    for l in lost:
        counts[l] = 0
    for r in reserve:
        counts[r] += 1
    
    for i in range(1, len(counts)-1):
        if counts[i] >0:
            answer += 1
        elif counts[i-1] == 2:
            answer += 1
        elif counts[i+1] == 2:
            answer += 1
            counts[i+1] = 1

    return answer
